### DISCUSSION TITLE
Adding additional L2-variables

### DISCUSSION THREAD

**Discussion by nrblsffspc** [jazznbass/scan#88] [Q&A] [answered]
Dear Community and Team,

in addition to the single case data for various children, we have collected the disruptive behavior of these children in a screening. Each child can therefore be assigned different raw values regarding the disruptive behavior. From this, we included two variables in the analyses: 

1) risk = risk status coded as 0 or 1 
2) risk_ges =  raw value of disruptive behavior coded as 0 - 50 

We have 4 different phases with the baseline (A,B,C,D) and now we want to find out which influence the risk status (0 or 1) or the total raw score has on the effectiveness of the support in the individual phases. 

Chapter 8 of the book explains in detail how the variables are inserted. This also works. Only here the example in the book is made with "sexm". But in the R code the variable is just called "sex". 

For our code this would mean that we are looking for the effect of "risk1" analogous to "sexm". So how do we have to proceed in the best case to adapt the code or the data in such a way that we can make statements about the influence of the risk status or the total value of the risk variable? 

Thanks for your help! 

Best regards

**Comment by jazznbass** [comment, marked as answer]
Hi,

you can just handle the risk and risk_ges variables as described in the scan book by adding it with the update argument of the hplm function.
The more complex answer is, that in my example, sex is a categorical variable of the type "factor" with two levels (m and f). I set the contrasts to treatment coding. That is, female was the reference category and "sexm" is the effect of the category "m" (in contrast to "f").
When you have a dichotomous variable with 0 and 1, 0 is automatically the reference category.
If you want it to appear as "risk1" you would have to turn "risk" into a factor (with the levels 0 and 1). By default, the treatment contrast would be set by R.

For you "risk_ges" variable, it is a bit more complicated.
If you just include risk_ges, the intercept and all other parameters will be estimated for a risk_ges value of  zero. That may be fine, but very often people want to have the estimates of the model based on the mean of a continuous variable.
Therefore, the variables are mean centered.

If your risk_ges is already included in an scdf, use the transform function (use ?transform.scdf to call the help page).
in your case, it would look something like this:
`my_scdf <- transform(my_scdf, risk_ges = scale(risk_ges, scale = FALSE))`

if it is in a separate file:
`my_l2data$risk_ges <- scale(my_l2data$risk_ges, scale = FALSE)`

Do not forget to add the interaction effect: 
`phaseB:risk + phaseB:risk_ges + phaseC:risk + phaseC:risk_ges` ... 
the exact syntax depends on whatever effect(s) you are interested in.

I hope, this helps,
Jürgen

**Reply by nrblsffspc** [reply]
Thanks so much!! That helps a lot and got things clear.
